common/src/main/java/io/druid/collections/OrderedMergeSequence.java

```diff
@@ -78,7 +87,23 @@ public OrderedMergeSequence(
   @Override
   public <OutType> Yielder<OutType> toYielder(OutType initValue, YieldingAccumulator<OutType, T> accumulator)
   {
-    PriorityQueue<Yielder<T>> pQueue = new PriorityQueue<Yielder<T>>(
+    return toYielder(Suppliers.ofInstance(initValue), accumulator);
+  }
+
+  @Override
+  public <OutType> Yielder<OutType> toYielder(
+      Supplier<OutType> initValue, YieldingAccumulator<OutType, T> accumulator
```

Comment:
suggest renaming `initValue` in the methods with supplier parameters to `initValueSupplier` to make it easier to distinguish the methods